pkg/cluster/ipaddresses.go

```diff
@@ -236,12 +236,21 @@ func (m *manager) ensureGatewayCreate(ctx context.Context) error {
 		return errors.New("private endpoint connection not found")
 	}
 
+	accountName, err := m.imageRegistryStorageAccountName()
+	if err != nil {
+		return err
+	}
+
+	if accountName == "" {
+		return fmt.Errorf("cluster manager's storage account name of image registry config is empty")
+	}
```

Comment:
Just a nit on the error message as well to be more concise. 
```suggestion
	if accountName == "" {
		return fmt.Errorf("the cluster's image registry name is empty")
	}
```

I believe the error message should be propagated to our logging system which should enrich the logger with the resource ID of the cluster in question which threw an error.  Can you confirm this, otherwise we should add the clusterId to the error message.

Reply:
About
> should we bundle this condition within the imageRegistryStorageAccountName() function? We're doing this in two places so would adhere to the DRY principal.

At the beginning, I created a function that did exactly that: get data and validate the data. 

Soon I realized that having a function that both gets data and at the same time validates it was a bad idea because there is a case where we want to get the data and if that data is not empty, just return no error. 

I am talking about pkg/cluster/storageaccounts.go:112
```go
accountName, err := m.imageRegistryStorageAccountName()
if err != nil {
	return err
}

if accountName != "" {
	return nil
}
```

If I have a function that both gets the data and validates it, when calling it in the piece of code I have shared, I will need to get the error, check the error == "cluster manager's storage account name of image registry config is empty" and bypass the error and do business logic depending on the error. In other words, the fact that imageRegistryStorageAccount == "" is an error just depends on the context. 

The best option would be to create a function validateImageRegistryStorageAccountName() and call it from the two places you mention.

About 
> I believe the error message should be propagated to our logging system which should enrich the logger...

I will change the error message as you suggest and check the logging.

Thanks!